src/traces/pie/defaults.js

```diff
@@ -13,21 +13,52 @@ var attributes = require('./attributes');
 var handleDomainDefaults = require('../../plots/domain').defaults;
 var handleText = require('../bar/defaults').handleText;
 
-module.exports = function supplyDefaults(traceIn, traceOut, defaultColor, layout) {
+function handleLabelsAndValues(labels, values) {
+    var hasLabels = Array.isArray(labels);
+    var hasValues = Lib.isArrayOrTypedArray(values);
+    var len = Math.min(
+        hasLabels ? labels.length : Infinity,
+        hasValues ? values.length : Infinity
+    );
+
+    if(!isFinite(len)) len = 0;
+
+    if(len && hasValues) {
+        var sum = 0;
+        for(var i = 0; i < len; i++) {
+            var v = +values[i];
+            if(v < 0) {
+                sum = 0;
+                break;
```

Comment:
Hmm. Don't you mean `continue;` here instead of `break;` ?

If I understand correctly, this makes any trace with one (or more) negative values has `_length=0` ?

Reply:
Good call. Done in https://github.com/plotly/plotly.js/pull/4416/commits/1967fd459ef07a8e09edc52a997d74539bd9914a.